src/main/java/com/joeyfrazee/nifi/reporting/ElasticsearchProvenanceReporter.java

```diff
@@ -56,21 +56,23 @@ public class ElasticsearchProvenanceReporter extends AbstractProvenanceReporter
             .displayName("Elasticsearch Index")
             .description("The name of the Elasticsearch index")
             .required(true)
-            .defaultValue("nifi")
+            .defaultValue(PluginEnvironmentVariable.ELASTICSEARCH_INDEX.getValue().orElse("nifi"))
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .build();
 
     public static final PropertyDescriptor ELASTICSEARCH_CA_CERT_FINGERPRINT = new PropertyDescriptor
             .Builder().name("Elasticsearch CA Certificate Fingerprint")
             .displayName("Elasticsearch CA Certificate Fingerprint")
             .description("The HTTP CA certificate SHA-256 fingerprint for Elasticsearch")
+            .defaultValue(PluginEnvironmentVariable.ELASTICSEARCH_CA_CERT_FINGERPRINT.getValue().orElse(null))
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Comment:
Does the `NON_EMPTY_VALIDATOR` fail if the value is empty/`null`?
I presume this will be empty if we're not using HTTPS?

Reply:
Also, we should improve the description stating it is only required when talking over HTTPS.

Reply:
> Does the `NON_EMPTY_VALIDATOR` fail if the value is empty/`null`? I presume this will be empty if we're not using HTTPS?

At least one validator must be specified: https://nifi.apache.org/docs/nifi-docs/html/developer-guide.html#validator

The `NON_EMPTY_VALIDATOR` prevents an empty string being set for the value, but doesn't prevent no value (null). Setting `required(true)` ensures a non-null value must be set.

There's a checkbox in the NiFi UI to specifically set empty strings for values. Checking the box will cause the validator to fail.